src/System.Text.Http/System/Text/Http/Parser/HttpUtilities.cs

```diff
@@ -152,110 +152,65 @@ public static string GetAsciiStringEscaped(this Span<byte> span, int maxChars)
         /// </remarks>
         /// <returns><c>true</c> if the input matches a known string, <c>false</c> otherwise.</returns>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static unsafe bool GetKnownMethod(this Span<byte> span, out Http.Method method, out int length)
+        public static bool GetKnownMethod(in this ReadOnlySpan<byte> span, out Http.Method method, out int length)
         {
-            fixed (byte* data = &MemoryMarshal.GetReference(span))
-            {
-                method = GetKnownMethod(data, span.Length, out length);
-                return method != Http.Method.Custom;
-            }
-        }
+            Debug.Assert(span.Length > sizeof(ulong));
 
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal unsafe static Http.Method GetKnownMethod(byte* data, int length, out int methodLength)
-        {
-            methodLength = 0;
-            if (length < sizeof(uint))
-            {
-                return Http.Method.Custom;
-            }
-            else if (*(uint*)data == _httpGetMethodInt)
-            {
-                methodLength = 3;
-                return Http.Method.Get;
-            }
-            else if (length < sizeof(ulong))
+            length = 0;
+            method = Http.Method.Custom;
+
+            if (Unsafe.ReadUnaligned<uint>(ref GetReference(span)) == _httpGetMethodInt)
```

Comment:
`MemoryMarshal.Read<uint>` ? Using the Unsafe class doesn't make the method fully safe (in release).

Reply:
This is just eliding the bounds check. Yes, it is technically unsafe, but I think it is worth going this far.